crates/sui-core/src/signature_mpc/identifiable_abort.rs

```diff
@@ -0,0 +1,181 @@
+    .unwrap();
+
+    failed_messages_indices
+        .iter()
+        .map(|index| {
+            generate_proof(
+                state.tiresias_public_parameters.clone(),
+                decryption_key_share.clone(),
+                state.party_id,
+                state.presigns.clone().unwrap().get(*index).unwrap().clone(),
+                state
+                    .tiresias_public_parameters
+                    .encryption_scheme_public_parameters
+                    .clone(),
+                state
+                    .public_nonce_encrypted_partial_signature_and_proofs
+                    .clone()
+                    .unwrap()
+                    .get(*index)
+                    .unwrap()
+                    .clone(),
+            )
+        })
+        .collect()
+}
+
+/// Identify all the parties that behaved maliciously in this messages batch.
+pub(crate) fn identify_batch_malicious_parties(
+    state: &SignState,
+) -> twopc_mpc_protocols::Result<HashSet<PartyID>> {
+    // Need to call [`generate_proofs`] to re-generate the SignaturePartialDecryptionProofVerificationParty objects,
+    // that are necessary to call the [`identify_malicious_parties`] function.
+    let failed_messages_parties = generate_proofs(&state, &state.failed_messages_indices.clone().unwrap());
+    let mut malicious_parties = HashSet::new();
+    let involved_shares = get_involved_shares(&state);
+    for ((i, message_index), (_, party)) in state
+        .clone()
+        .failed_messages_indices
+        .unwrap()
+        .into_iter()
+        .enumerate()
+        .zip(failed_messages_parties.into_iter())
+    {
+        let (shares, masked_shares) = change_shares_type(&involved_shares, message_index);
+        let involved_proofs = get_involved_proofs(&state, i);
+        identify_message_malicious_parties(
+            party,
+            shares,
+            masked_shares,
+            state.tiresias_public_parameters.clone(),
+            involved_proofs,
+            state
+                .involved_parties
+                .as_deref()
+                .unwrap_or(&Vec::new())
+                .into(),
+        )
+        .iter()
+        .for_each(|party_id| {
+            malicious_parties.insert(*party_id);
+        });
+    }
+    Ok(malicious_parties)
+}
+
+/// Maps the decryption shares to the type expected by the identify_malicious_decrypters function from the 2pc-mpc repository.
+fn change_shares_type(
+    involved_shares: &HashMap<
+        PartyID,
+        Vec<(PaillierModulusSizedNumber, PaillierModulusSizedNumber)>,
+    >,
+    message_index: usize,
+) -> (
+    HashMap<PartyID, PaillierModulusSizedNumber>,
+    HashMap<PartyID, PaillierModulusSizedNumber>,
+) {
+    involved_shares
+        .iter()
+        .map(|(party_id, shares)| {
+            (
+                (*party_id, shares[message_index].0.clone()),
+                (*party_id, shares[message_index].1.clone()),
+            )
+        })
+        .unzip()
```

Comment:
if this logic doesn't repeat in two different locations, we don't need a function for this.